File: demo-android/app/src/androidTest/java/com/example/demo/e2e/interaction/page/EditTaskForm.kt
package com.example.demo.e2e.interaction.page

import androidx.compose.ui.test.junit4.AndroidComposeTestRule
import com.example.demo.MainActivity
import com.example.demo.data.model.TaskPriority
import com.example.demo.data.model.TaskStatus

class EditTaskForm(
    rule: AndroidComposeTestRule<*, MainActivity>,
) : ComposePage(rule) {
    fun saveButton() = node("save-button")

    fun titleField() = node("edit-task-title-input")

    fun descriptionField() = node("task-description-input")

    fun statusDropdown() = node("status-dropdown")

    fun priorityDropdown() = node("priority-dropdown")

    fun statusOption(status: TaskStatus) = node("status-dropdown-option-${status.name}")

    fun priorityOption(priority: TaskPriority) = node("priority-dropdown-option-${priority.name}")
}

File: demo-ios/DemoUITests/Interaction/Page/EditTaskForm.swift
import XCTest

final class EditTaskForm: XCUIPage {
    func saveButton() -> XCUIElement {
        element("save-button")
    }

    func titleField() -> XCUIElement {
        element("edit-task-title-input")
    }

    func descriptionField() -> XCUIElement {
        element("task-description-input")
    }

    func statusDropdown() -> XCUIElement {
        element("status-dropdown")
    }

    func priorityDropdown() -> XCUIElement {
        element("priority-dropdown")
    }

    func statusOption(_ status: TaskStatus) -> XCUIElement {
        element("status-dropdown-option-\(status.rawValue)")
    }

    func priorityOption(_ priority: TaskPriority) -> XCUIElement {
        element("priority-dropdown-option-\(priority.rawValue)")
    }
}
